beacon-chain/state/stategen/getter.go

```diff
@@ -109,6 +113,25 @@ func (s *State) StateBySlot(ctx context.Context, slot types.Slot) (*state.Beacon
 	return s.loadStateBySlot(ctx, slot)
 }
 
+// StateByStateRoot retrieves the state with root equal to the input root.
+// It searches for the state only in the last SLOTS_PER_HISTORICAL_ROOT state roots.
+func (s *State) StateByStateRoot(
```

Comment:
does this method need to be in `stategen` ? seems like you could have it in the `rpc/beaconV1` package .

Reply:
You're right, it does not use anything internal to `stategen`.